packages/web-application/facades/add-cname-records.ts

```diff
@@ -0,0 +1,22 @@
+import { WebApplication } from '../web-application.construct';
+import { CnameRecord } from '@aws-cdk/aws-route53';
+import { pascalCase } from 'change-case';
+import { DOMAIN_NAME_REGISTRAR } from '../constants';
+import { CloudFrontWebDistribution } from '@aws-cdk/aws-cloudfront';
+
+export function addCnameRecords(
+  this: WebApplication,
```

Comment:
Adding typing to `this` on order to use `call()` seems like a really confusing pattern. Why not just pass in the application as an argument?

Reply:
In fact, if all you need is the props, why not just pass in the props?

Reply:
basically I want to make sure that `this` in here refers back to web-application.construct.ts, but you are right this does seem bit confusing.